src/Microsoft.ML.Transforms/Text/NgramTransform.cs

```diff
@@ -597,12 +597,12 @@ private void GetSlotNames(int iinfo, int size, ref VBuffer<ReadOnlyMemory<char>>
             {
                 Host.Assert(0 <= iinfo && iinfo < _parent.ColumnPairs.Length);
 
-                var keyCount = _srcTypes[iinfo].GetItemType().GetKeyCount();
-                Host.Assert(InputSchema[_srcCols[iinfo]].HasKeyValues(keyCount));
+                Host.Assert(InputSchema[_srcCols[iinfo]].HasKeyValues(_srcTypes[iinfo].GetItemType()));
 
                 var unigramNames = new VBuffer<ReadOnlyMemory<char>>();
 
                 // Get the key values of the unigrams.
+                var keyCount = _srcTypes[iinfo].GetItemType().ChecktRangeReturnCount(Host);
```

Comment:
(nit) why not capture `_srcTypes[iinfo].GetItemType()` in a variable so it doesn't need to be called twice?

Reply:
Done!